src/lightning/fabric/strategies/deepspeed.py

```diff
@@ -339,6 +339,11 @@ def setup_optimizer(self, optimizer: Optimizer) -> Optimizer:
         """
         raise NotImplementedError(self._err_msg_joint_setup_required())
 
+    @contextmanager
+    def module_init_context(self) -> Generator[None, None, None]:
+        with self.module_sharded_context():
```

Comment:
Isn't this missing the precision call?

Reply:
Yes to some degree. DeepSpeed's Zero context manager handles precision already. So in theory we should put the precision context inside the else that mirrors `self.zero_stage_3` in L355. For now, I added it conditionally in the init_module method, but in the future, we may want to rearrange the logic for better readability.